Add --list flag to display module upgrades without interactivity

bneigher opened the pull request:
## Summary
Adds a new `--list/-l` CLI flag that displays available module upgrades without any interactivity, using the same color formatting as the interactive mode.

## Changes
- Add new CLI flag `--list/-l` to list available module upgrades
- Implement `listModules()` function using existing color formatting logic
- Support same color coding as interactive mode:
  - Red for major version 0 (unstable modules)
  - Yellow for minor version upgrades  
  - Green for patch version upgrades
  - Blue for current version formatting
- Update control flow to handle list mode alongside force and interactive modes
- Maintains compatibility with existing flags (verbose, ignore, etc.)

## Usage
```bash
# List all available upgrades with colors
go-mod-upgrade --list

# Can be combined with other flags
go-mod-upgrade --list --verbose
go-mod-upgrade --list --ignore "example.com"
```

## Testing
- [x] Code compiles successfully
- [x] Help message displays the new flag correctly
- [x] No linting errors introduced

This feature provides a quick, non-interactive way to see what module upgrades are available, which is useful for CI/CD pipelines, scripts, or when you just want to check what's available without going through the interactive selection process.

Commit:
Add --list flag to display module upgrades without interactivity

- Add new CLI flag --list/-l to list available module upgrades
- Implement listModules() function using existing color formatting logic
- Support same color coding as interactive mode:
  - Red for major version 0 (unstable)
  - Yellow for minor version upgrades
  - Green for patch version upgrades
  - Blue for current version formatting
- Update control flow to handle list mode alongside force and interactive modes
- Maintains compatibility with existing flags (verbose, ignore, etc.)

Commit:
Update README with --list flag documentation

- Add --list flag to global options section
- Add new section with usage examples for the list functionality
- Document that --list can be combined with other flags like --verbose and --ignore
- Explain that --list is perfect for CI/CD pipelines and non-interactive usage

Commit:
clean up list readme entry

oligot commented:
Hi,

This is a really great contribution, thanks !
I just activated the checks and 2 of them failed: Lint and Vulnerability.
Do you see how to fix them ?

Commit:
Fix linting and vulnerability issues

- Fix errcheck linting issue by properly handling fmt.Fprintf error return value in listModules function
- Upgrade Go version from 1.23.x to 1.25.1 to resolve vulnerability GO-2025-3956 in os/exec package
- Add proper toolchain specification to go.mod
- Follow existing error handling patterns from the codebase

bneigher commented:
I think that should do it. Can you try the lint and vulnerability checks again?

oligot committed:
fix: lint and vulncheck issues

By upgrading nixpkgs as the "current" versions are not compatible with
Go 1.25

oligot commented:
I had to update the nixpkg but now it works :tada:

oligot commented:
Manually squashed and merged: 920a076992f8e05cbb30e9759de7cbe6271688c5 and f2840960bdfca0a90ff2da44f1b0189039689be3

bneigher commented:
thank you.  can you publish a new tag for it? I want to try it in my repo's CI/CD job 🙏

oligot commented:
[v0.12.0](https://github.com/oligot/go-mod-upgrade/releases/tag/v0.12.0) just released :tada:
